### PR TITLE
x-enum-varnames revival

### PR DESCRIPTION
**Pull request by amery**
This is https://github.com/deepmap/oapi-codegen/pull/692 by @reedom rebased on top of master

### CODE REVIEW

File: pkg/codegen/test_spec.yaml [hunk 1]
```diff
@@ -136,3 +157,23 @@ components:
...
+
+    EnumTest:
+      properties:
+        numerics:
+          type: integer
+          enum: [0, 1, 2]
+        enumNames:
+          type: integer
+          enum: [0, 1, 2]
+          x-enum-varnames:
```
Comment (derfenix) [inline pkg/codegen/test_spec.yaml:169]: `x-enumNames` here?

**Comment by amery** [comment]
rebased up 1.12.4

**Comment by Antonboom** [comment]
@deepmap-marcinr

**Comment by amery** [comment]
thank you @deepmap-marcinr!

**Comment by saulhoward** [comment]
Great feature! In my testing, this removes any underscores in the provided enum names -- I assume this is a bug?

**Pull request merged**